### PR TITLE
Make ubuntu 1604 helix queue required PR check

### PR DESCRIPTION
**Pull request by HaoK**
- Adds ubuntu 1604 to required PR checks
- Stops submitting reuqired helix jobs to queues twice (helix-test job won't run required checks anymore)
- switch to different arm queue to reduce load on a helix queue

### CODE REVIEW

File: .azure/pipelines/ci.yml [hunk 1]
```diff
@@ -655,4 +674,4 @@ stages:
       # See https://github.com/dotnet/arcade/issues/2871
       enableSymbolValidation: false
       enableSigningValidation: false
-      publishInstallersAndChecksums: true
+      publishInstallersAndChecksums: true
```
Comment (dougbu) [inline .azure/pipelines/ci.yml]: Don't change the final newline if you can help it. Most prefer they are present and leaving the files alone prevents thrashing back and forth.

**Review by dougbu — COMMENTED** [review]
@Pilchie should this be going into 'release/3.1' instead?

**Comment by Pilchie** [comment]
Do we think this will make our 3.1 test runs *more stable*?  It doesn't seem like we're at that point right now, and it doesn't seem like we're going to get to the point of running the majority of our tests only in Helix without significant changes to the build, which I'm not sure we want to do in the 3.1 branches.  Though it pains me not to have it for LTS, I think that the majority of our Helix work probably should happen in master.

Now tell me why I'm wrong... 😉

**Comment by HaoK** [comment]
Unfortunately I agree with you @Pilchie making this a required PR check will likely only cause additional tests to be red, and doesn't really gain us anything in release/3.1.  I think master is more appropriate as this is more of a baby step to see how reliable the unbuntu queue is, where if it is, we can maybe start considering how close we are to being able to retire the ubuntu ci test job in favor of the helix check

**Comment by dougbu** [comment]
Nobody's wrong. I just wanted to air the question.

Think we've got a worse issue on the Helix ARM64 side in any case. That's still just in 'master' but we seem (per @jashook) to have taken enough capacity to overload that Helix queue.

**Comment by jashook** [comment]
I doubt there is enough volume of work generated in the release branch to overload any queue.

**Comment by HaoK** [comment]
Well we were already using these queues in master, but it does look like we are using them 2x per PR, I can fix that now, and add logic so we don't hit the same queue in both the aspnet-ci job and helix-test jobs

**Comment by jashook** [comment]
As a follow up question, how often does aspnet expect to break arm, but not break x64?

**Comment by jashook** [comment]
ping @HaoK

**Comment by HaoK** [comment]
We were asked in the past to make sure we had arm coverage for our tests, which was why we added these queues in the first place, it was an external ask for us, but yes in general we don't expect this to catch anything

**Comment by jashook** [comment]
It seems like a correct definition to having arm coverage in aspnet to me would be having rolling or batch ci runs which include arm64 test runs. That way you can give the lower stack more coverage on the platform, while also not taking too much hardware resources.

**Comment by jashook** [comment]
PR testing on arm hardware, when there is little expectation to break a specific architecture has non trivial cost, for little perceived benefit.

**Review by dougbu — APPROVED** [review]

**Comment by dougbu** [comment]
@jashook perhaps you're right longer term. But, this PR immediately goes back to just one Helix ARM64 per PR. Including that in _just_ rolling builds may be a good idea on top of that. We'll discuss.

**Comment by jashook** [comment]
@dougbu I think this issue is more deeply rooted, as changing to the armarch queue is interesting/useful, only if we expect to run arm64 testing per PR. Note that that queue also has its own amount of hardware utilization, which this may overload as well. The suggestion to move to it was under the assumption that testing arm64 on PR is a necessity.

**Comment by Pilchie** [comment]
@jashook we'll discuss and figure out a plan moving forward.  I'm not tied to running ARM64 jobs on PR, but it's not a change for us, we've been doing it for a very long time.  What changed recently is that we started running it twice per PR, which @HaoK has now fixed.

I'm open to the idea of running it as an outerloop type test, but at the moment, this repo doesn't have any of those, and doesn't run Helix anywhere *except* on PRs, so it's not a trivial thing to move it there.

**Comment by Pilchie** [comment]
@jashook we'll discuss and figure out a plan moving forward.  I'm not tied to running ARM64 jobs on PR, but it's not a change for us, we've been doing it for a very long time.  What changed recently is that we started running it twice per PR, which @HaoK has now fixed.

I'm open to the idea of running it as an outerloop type test, but at the moment, this repo doesn't have any of those, and doesn't run Helix anywhere *except* on PRs, so it's not a trivial thing to move it there.

**Pull request merged**